src/Pos/Context/Functions.hs

```diff
@@ -83,3 +85,11 @@ recoveryInProgress :: (MonadIO m, MonadRecoveryHeader ssc m) => m Bool
 recoveryInProgress = do
     var <- Ether.ask @RecoveryHeaderTag
     isJust <$> atomically (STM.tryReadTMVar var)
+
+
+
+recoveryCommGuard
```

Comment:
1. Three empty lines.
2. No documentation.
3. Misplacement: you've just added import of `Pos.Communication.Protocol` into `Context.Functions`. `Context.*` are supposed to be lightweight types and not import anything heavy. Another file -- `Context.Context` does some imports from `Communication`, but they are type-oriented. Solution: import `Protocol.Types` instead (if it's possible, idk) or move `recoveryCommGuard` elsewhere.